src/Microsoft.Data.Analysis/DataFrame.cs

```diff
@@ -463,35 +463,42 @@ public DataFrame Append(IEnumerable<object> row = null, bool inPlace = false)
             bool columnMoveNext = columnEnumerator.MoveNext();
             if (row != null)
             {
-                    // Go through row first to make sure there are no data type incompatibilities
-                    IEnumerator<object> rowEnumerator = row.GetEnumerator();
-                    bool rowMoveNext = rowEnumerator.MoveNext();
-                    List<object> cachedObjectConversions = new List<object>();
-                    while (columnMoveNext && rowMoveNext)
+                // Go through row first to make sure there are no data type incompatibilities
+                IEnumerator<object> rowEnumerator = row.GetEnumerator();
+                bool rowMoveNext = rowEnumerator.MoveNext();
+                List<object> cachedObjectConversions = new List<object>();
+                while (columnMoveNext && rowMoveNext)
+                {
+                    DataFrameColumn column = columnEnumerator.Current;
+                    object value = rowEnumerator.Current;
+                    // StringDataFrameColumn can accept empty strings. The other columns interpret empty values as nulls
+                    if (value is string stringValue)
                     {
-                        DataFrameColumn column = columnEnumerator.Current;
-                        object value = rowEnumerator.Current;
-                        // StringDataFrameColumn can accept empty strings. The other columns interpret empty values as nulls
-                        if (value is string stringValue && string.IsNullOrEmpty(stringValue) && column.DataType != typeof(string))
+                        if (string.IsNullOrEmpty(stringValue) && column.DataType != typeof(string))
```

Comment:
You know `stringValue` is not null here, so you can change `string.IsNullOrEmpty(stringValue)` => `stringValue.Length == 0`.